skmultilearn/model_selection/iterative_stratification.py

```diff
@@ -0,0 +1,207 @@
+          address =      {ECML-PKDD, Skopje, Macedonia},
+          publisher =    {PMLR},
+        }
+"""
+
+
+
+from sklearn.model_selection._split import _BaseKFold
+import numpy as np
+import scipy.sparse as sp
+import itertools
+
+
+class IterativeStratification(_BaseKFold):
+    """Iteratively stratify a multi-label data set
+
+    Construct an interative stratifier that conducts:
+
+    :param n_splits: with number of splits
+    :type n_splits: int
+
+    :param order: taking into account order-th order of relationship
+    :type order: int
+
+    :param random_state: and the random state seed (optional)
+    :type random_state: int
+
+    """
+    def __init__(self, n_splits=3, order=1, random_state=None):
+        self.order = order
+        super(
+            IterativeStratification,
+            self).__init__(n_splits,
+                           shuffle=False,
+                           random_state=random_state)
+
+    def init_values(self, X):
+        self.n_samples = X.shape[0]
+        self.n_labels = X.shape[1]
+        self.percentage_per_fold = [1 / float(self.n_splits)
+                                    for i in range(self.n_splits)]
+        self.desired_samples_per_fold = np.array([self.percentage_per_fold[i] * self.n_samples
+                                                  for i in range(self.n_splits)])
+        self.rows = sp.lil_matrix(X).rows
+        self.rows_used = {i: False for i in range(self.n_samples)}
+        
+        self.folds = [[] for i in range(self.n_splits)]
+        
+        self.init_row_and_label_data(X)
+
+    def init_row_and_label_data(self, X):
+        self.all_combinations = []
+        self.per_row_combinations = [[] for i in range(self.n_samples)]
+        self.samples_with_combination = {}
+        for i, x in enumerate(self.rows):
+            for y in itertools.combinations_with_replacement(x, self.order):
+                if y not in self.samples_with_combination:
+                    self.samples_with_combination[y] = []
+
+                self.samples_with_combination[y].append(i)
+
+                self.all_combinations.append(y)
+                self.per_row_combinations[i].append(y)
+
+        self.all_combinations = [list(x) for x in set(self.all_combinations)]
+
+        self.desired_samples_per_label_per_fold = {
+            i:
+               [len(self.samples_with_combination[i]) * self.percentage_per_fold[j]
+                for j in range(self.n_splits)]
+                for i in self.samples_with_combination
+        }
+
+    def label_tie_break(self, M):
+        if len(M) == 1:
+            return M[0]
+        else:
+            max_val = max(self.desired_samples_per_fold[M])
+            M_prim = np.where(
+                np.array(self.desired_samples_per_fold) == max_val)[0]
+            M_prim = np.array([x for x in M_prim if x in M])
+            return np.random.choice(M_prim, 1)[0]
+
+    def get_most_desired_combination(self):
```

Comment:
I think this can be written in such a way that `currently_chosen` is always returned instead of having two return statements. Since the initial value of `currently_chosen` is `None`, then we just return this initial value given these conditions and return another value given another set of conditions. I think this may be a better practice than returning the actual `None` itself.

This means that everything inside the `for-loop` corresponds to changing the value of whatever `currently_chosen` is.

Reply:
done.